scripts/modules/elastic_stack.py

```diff
@@ -520,22 +414,21 @@ def _content(self):
         for param, value in self.apm_server_command_args:
             command_args.extend(["-E", param + "=" + value])
 
-        healthcheck_path = "/" if self.at_least_version("6.5") else "/healthcheck"
-
         ports = [
             self.publish_port(self.port, self.SERVICE_PORT),
             self.publish_port(self.apm_server_monitor_port, self.DEFAULT_MONITOR_PORT),
         ]
-        if ("apm-server.jaeger.http.enabled", "true") in self.apm_server_command_args:
-            ports.append(self.publish_port(self.DEFAULT_JAEGER_HTTP_PORT))
-
-        if ("apm-server.jaeger.grpc.enabled", "true") in self.apm_server_command_args:
-            ports.append(self.publish_port(self.DEFAULT_JAEGER_GRPC_PORT))
 
         command = ["apm-server", "-e", "--httpprof", ":{}".format(self.apm_server_monitor_port)] + command_args
         if self.options.get("apm_server_enable_debug"):
             command = command + ["-d", "*"]
 
+        healthcheck_path = "/" if self.at_least_version("6.5") else "/healthcheck"
+        healthcheck = curl_healthcheck(self.SERVICE_PORT, path=healthcheck_path)
+        if self.at_least_version("8.0"):
+            curl_command = "curl -s http://localhost:{}/ | grep -q 'publish_ready.*true'".format(self.SERVICE_PORT)
+            healthcheck["test"] = ["CMD-SHELL", curl_command]
```

Comment:
We can check this from `7.16` onwards, correct?

Reply:
We _can_, but it's not necessary - we're only enabling data streams by default for 8.0+. The server is "publish ready" by default in 7.16.